From 5ea68061f177da477dd11c649720110e8d8d7351 Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Fri, 20 Apr 2018 18:19:38 -0500
Subject: [PATCH] watchdog: Fix SPL builds when watchdog is enabled Without
 this fix, u-boot fails to comple as it tries to load watchdog.h during an SPL
 build. This breaks support for things such as the sunxi arch.

---
 include/watchdog.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/watchdog.h b/include/watchdog.h
index 52f4c506b0..614bf0e96a 100644
--- a/include/watchdog.h
+++ b/include/watchdog.h
@@ -21,7 +21,7 @@
 int init_func_watchdog_reset(void);
 #endif
 
-#if defined(CONFIG_WATCHDOG) || defined(CONFIG_HW_WATCHDOG)
+#if (defined(CONFIG_WATCHDOG) || defined(CONFIG_HW_WATCHDOG)) && !defined(CONFIG_SPL_BUILD)
 #define INIT_FUNC_WATCHDOG_INIT	init_func_watchdog_init,
 #define INIT_FUNC_WATCHDOG_RESET	init_func_watchdog_reset,
 #else
@@ -36,7 +36,7 @@ int init_func_watchdog_reset(void);
 /*
  * Hardware watchdog
  */
-#ifdef CONFIG_HW_WATCHDOG
+#if defined(CONFIG_HW_WATCHDOG) && !defined(CONFIG_SPL_BUILD)
 	#if defined(__ASSEMBLY__)
 		#define WATCHDOG_RESET bl hw_watchdog_reset
 	#else
@@ -48,7 +48,7 @@ int init_func_watchdog_reset(void);
 	/*
 	 * Maybe a software watchdog?
 	 */
-	#if defined(CONFIG_WATCHDOG)
+	#if defined(CONFIG_WATCHDOG) && !defined(CONFIG_SPL_BUILD)
 		#if defined(__ASSEMBLY__)
 			#define WATCHDOG_RESET bl watchdog_reset
 		#else
